clevertap-core/src/main/java/com/clevertap/android/sdk/utils/CTJsonConverter.java

```diff
@@ -90,6 +90,13 @@ public static JSONObject from(DeviceInfo deviceInfo, CoreMetaData coreMetaData,
             evtData.put("hgt", deviceInfo.getHeight());
             evtData.put("dpi", deviceInfo.getDPI());
             evtData.put("dt", DeviceInfo.getDeviceType(deviceInfo.getContext()));
+
+            // If locale is set by the client then use that, otherwise fetch it from the device
+            String locale = deviceInfo.getCustomLocale();
+            if(TextUtils.isEmpty(locale))
+                locale = deviceInfo.getLocale();
```

Comment:
if locale is empty we get it from device, but you have created a setter method that allows setting emply locale. Can we make the change in setter itself so empty is not a valid value?

Reply:
We wanted this behaviour similar to leanplum and hence there is no validation on the locale provided in setCustomLocale. Here we expect the user to send an appropriate locale, it can be of any format